### PR TITLE
feat(TG-395): constructor map view

### PR DESCRIPTION
**Pull request by borys-ihnatyev**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-frontend/pull/184" target="_blank">ReviewNinja

File: src/app/constructor/map/map-constructor.component.ts [hunk 1]
```diff
@@ -1,40 +1,138 @@
...
+            .subscribe((mapViewInfoDto) => {
+                this.uploader.setOptions({
+                    url: this.mapViewService.resolvePictureUploadUrl(mapViewInfoDto)
+                });
+                this.uploader.uploadAll();
+            });
+    }
+
+    private isMapViewCanBeSaved(): boolean {
+        return !!this.mapViewName && !!this.mapViewDescription && !!this.picture;
```
Comment (yakimko) [inline src/app/constructor/map/map-constructor.component.ts]: return this.mapViewName && this.mapViewDescription && this.picture;

File: src/app/constructor/devices/devices.component.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    .map(ref => ref.nativeElement);
+                this.drawSensor();
+            });
+
+        this.dragAndDrop();
+    }
+
+    drawSensor(): void {
+        if (this.sensors.length) {
+            let len = this.deviceList.length;
```
Comment (velmyk) [inline src/app/constructor/devices/devices.component.ts:40]: All this values could be `const`

File: src/app/constructor/map/map-constructor.component.ts (hunk 1 above)
Comment (borys-ihnatyev) [inline src/app/constructor/map/map-constructor.component.ts]: !fixed

File: src/app/constructor/devices/devices.component.ts (hunk 2 above)
Comment (borys-ihnatyev) [inline src/app/constructor/devices/devices.component.ts:40]: !fixed

**Pull request merged**